src/hip_stream.cpp

```diff
@@ -254,14 +254,101 @@ hipError_t hipStreamGetPriority(hipStream_t stream, int* priority) {
 
 
 //---
+hipError_t setCallbackPacket(hsa_queue_t* queue,
+                             uint64_t& index, uint64_t& nextIndex,
+                             hsa_barrier_and_packet_t** barrier1,
+                             hsa_barrier_and_packet_t** barrier2){
+
+    uint64_t tempIndex = 0;
+    uint32_t mask = queue->size - 1;
```

Comment:
I'm not 100% up to date on the HIP runtime's coding style, but should you verify that queue != null before dereferencing it?